frontend/src/common/UserDisplay.tsx

```diff
@@ -0,0 +1,93 @@
+import EmailIcon from '@mui/icons-material/Email'
+import UserIcon from '@mui/icons-material/Person'
+import { Box, Divider, Popover, Stack, Typography } from '@mui/material'
+import { useListUsers } from 'actions/user'
+import { MouseEvent, useEffect, useState } from 'react'
+import Loading from 'src/common/Loading'
+import { EntityObject } from 'types/v2/types'
+
+type ConditionalUserDisplayProps =
+  | {
+      entityId?: string
+      entity?: never
+    }
+  | {
+      entityId?: never
+      entity?: EntityObject
+    }
+
+type CommonUserDisplayProps = {
+  hidePopover?: boolean
+}
+
+type UserDisplayProps = CommonUserDisplayProps & ConditionalUserDisplayProps
+
+export default function UserDisplay({ entity, entityId, hidePopover = false }: UserDisplayProps) {
+  const [user, setUser] = useState<EntityObject | undefined>(entity || undefined)
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
+  const open = Boolean(anchorEl)
+
+  const { users, isUsersLoading } = useListUsers(entityId ? entityId : '')
+
+  useEffect(() => {
+    if (!user && users) {
+      setUser(users.find((userFromList) => userFromList.id === entityId))
+    }
+  }, [entityId, users, user])
+
+  return (
+    <>
+      {isUsersLoading && <Loading />}
```

Comment:
This JSX could be a bit cleaner if we put the loading block outside of the main JSX block. E.g.
```
if (isUsersLoading || !user) {
  return <Loading />
}

return (
  // main JSX block doesn't need to do any isUsersLoading or user checks
)